trestle/core/profile_resolver.py

```diff
@@ -29,17 +33,49 @@ class ProfileResolver():
     """Class to resolve a catalog given a profile."""
 
     @staticmethod
-    def get_resolved_profile_catalog(
+    def _extract_inherited_props(res_cat: cat.Catalog) -> Tuple[cat.Catalog, Dict[str, Any]]:
+        """
+        Build the control dict of inherited props.
+
+        Args:
+            The resolved profile catalog with a possible temporary part in each control
+
+        Returns:
+            The temporary parts are removed from each control and a Dict of added props per control_id is returned
+
+        Notes:
+        If an upstream profile adds props to the control they are tracked in a special temporary part in the control
+        called const.TRESTLE_INHERITED_PROPS.  If that part is present in a control its contents should be added to
+        the dict entry for that control, and the part removed from the control.
+        """
+        prop_dict: Dict[str, Any] = {}
+        cat_interface = CatalogInterface(res_cat)
+        for control in cat_interface.get_all_controls_from_dict():
+            part: com.Part = pop_item_from_list(control.parts, TRESTLE_INHERITED_PROPS, lambda p: p.name)
+            if part:
+                props_list: List[Dict[str, Any]] = []
+                for prop in as_list(part.props):
+                    props_list.append({'name': prop.name, 'value': prop.value})
+                for sub_part in as_list(part.parts):
+                    for prop in as_list(sub_part.props):
+                        props_list.append({'name': prop.name, 'value': prop.value, 'part_name': sub_part.title})
```

Comment:
what about sub_part.part?

Reply:
This is a structure that trestle creates and it is not recursive.  It's just a way to keep track of added props.  Props are added either at the control level as a single list, or by_id as a list bound a by_id sub_part.  It is described in the docs of that function.

Reply:
Seems a bit limiting to not display props in the deepest parts, but if that is by design then okay.

Reply:
All added props are captured in this list - they can be added at any level - by-id - and they will appear in this list.  There is no need for a deep list because this is just a list of added props that trestle uses to feed them out to the catalog writer.  And then the list is deleted from the control.